### PR TITLE
integrate postgresql into the application

### PR DESCRIPTION
**Pull request by kartelpol**
https://epa.ms/nodejs18-hw6

### CODE REVIEW

File: src/controllers/authentication.js [hunk 1]
```diff
@@ -1,55 +1,69 @@
 import jwt from 'jsonwebtoken';
 import uuid from 'uuid';
 
-export function authenticationController(credentials, secretCode) {
-    return (req, res) => {
+export function authenticationController(secretCode) {
+    return async (req, res) => {
         const userInfo = req.body;
+        const userResponse = await req.context.models.User.findAll({
```
Comment (deser) [inline src/controllers/authentication.js]: It's nice to abstract DB operations with abstraction layer

File: src/controllers/authentication.js [hunk 2]
```diff
@@ -1,55 +1,69 @@
...
+            where: {
+                name: userInfo.name,
+                password: userInfo.password
```
Comment (deser) [inline src/controllers/authentication.js]: Do you store passwords not encrypted?
https://stackoverflow.com/questions/1054022/best-way-to-store-password-in-database

File: src/controllers/authentication.js [hunk 3]
```diff
@@ -1,55 +1,69 @@
...
+            }
+        });
 
-        if (credentials[userInfo.name].password === userInfo.password) {
-            const token = jwt.sign( {password: userInfo.password}, secretCode, { expiresIn: 30 });
-            const refreshToken = uuid();
+        const user = userResponse[0];
```
Comment (deser) [inline src/controllers/authentication.js]: userResponse is always defined and is array?

File: src/controllers/authentication.js [hunk 4]
```diff
@@ -1,55 +1,69 @@
...
                 refreshToken,
                 token,
             };
-            
-            res
-                .set({'X-access-token': token})
-                .json(message);
-            
+
+            user.refreshToken = refreshToken;
```
Comment (deser) [inline src/controllers/authentication.js]: can you re-use `refreshTokenController` here?

File: src/controllers/products.js [hunk 5]
```diff
@@ -1,18 +1,31 @@
 function getAll(req, res) {
-    res.send('ALL products');
+    req.context.models.Product.findAll()
```
Comment (deser) [inline src/controllers/products.js]: return?

File: src/controllers/products.js [hunk 6]
```diff
@@ -1,18 +1,31 @@
...
+    }, {})
+        .then(product => res.json(product))
+        .catch(err => res.send(err));
 }
 
 function getReviews(req, res) {
-    res.send(`Product's with id: ${req.params.id} reviews`);
+    req.context.models.Review.findAll({
+        where: {
+            productId: req.params.id
```
Comment (deser) [inline src/controllers/products.js]: getting request parameters should be also abstracted. For example in future you'd definitely like to add validation and that would be nice place to add it

File: src/middlewares/passport.js [hunk 7]
```diff
@@ -5,16 +5,20 @@ import Facebook from 'passport-facebook';
...
             passwordField: 'password'
         },
-        (username, password, done) =>  credentials[username] && credentials[username].password === password ?
-            done(null, credentials[username]) : done(null, false, "Basic")
+        (username, password, done) =>  user.findAll({
+            where: {
+                name: username,
+                password: password,
+            }
+        }).then(dbResponse => dbResponse[0] ? done(null, dbResponse[0] ) : done(null, false, "Basic"))
```
Comment (deser) [inline src/middlewares/passport.js:21]: use _.head

File: src/models/index.js [hunk 8]
```diff
@@ -1,7 +1,56 @@
...
+);
+
+const models = {
+  User: createUser(sequelize, Sequelize),
+  Product: createProduct(sequelize, Sequelize),
+}
+
+Object.keys(models).forEach(key => 'associate' in models[key] && models[key].associate(models));
+
+async function fillDB (models) {
```
Comment (deser) [inline src/models/index.js]: To separate file

Comment (deser) [reply src/models/index.js]: Also why do you need it when sequelize already has `seeders-path`?

File: src/postgres/models/index.js [hunk 9]
```diff
@@ -0,0 +1,49 @@
+'use strict';
```
Comment (deser) [inline src/postgres/models/index.js:1]: what is the difference between this file and src/models/index?

File: src/postgres/models/index.js [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+}
+const user = getUser(sequelize, Sequelize);
+const product = getProduct(sequelize, Sequelize);
+const review = getReview(sequelize, Sequelize);
+
+db['User'] = user;
+db['Product'] = product;
+db['Review'] = review;
+
+// introduce webpack dynamic imports
```
Comment (deser) [inline src/postgres/models/index.js:29]: remove

File: src/postgres/models/review.js [hunk 11]
```diff
@@ -0,0 +1,11 @@
+'use strict';
+module.exports = (sequelize, DataTypes) => {
```
Comment (deser) [inline src/postgres/models/review.js:2]: seems you duplicate these models

File: src/models/index.js (hunk 8 above)
Comment (kartelpol) [reply src/models/index.js]: Just forgot to delete

File: src/postgres/models/index.js (hunk 9 above)
Comment (kartelpol) [reply src/postgres/models/index.js:1]: this I generated with sequelize-cli, src/models/index was previously written by me

File: src/controllers/products.js (hunk 6 above)
Comment (kartelpol) [reply src/controllers/products.js]: resolve in the next task

**Pull request merged**